sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -0,0 +1,1248 @@
+
+        -'constant', is a constant learning rate given by
+         'learning_rate_init'.
+
+        -'invscaling' gradually decreases the learning rate 'learning_rate_' at
+          each time step 't' using an inverse scaling exponent of 'power_t'.
+          learning_rate_ = learning_rate_init / pow(t, power_t)
+
+        -'adaptive', keeps the learning rate constant to
+         'learning_rate_init' as long as the training keeps decreasing.
+         Each time 2 consecitive epochs fail to decrease the training loss by
+         tol, or fail to increase validation score by tol if 'early_stopping'
+         is on, the current learning rate is divided by 5.
+
+         Only used when algorithm='sgd'.
+
+    max_iter : int, optional, default 200
+        Maximum number of iterations. The algorithm
+        iterates until convergence (determined by 'tol') or
+        this number of iterations.
+
+    random_state : int or RandomState, optional, default None
+        State of or seed for random number generator.
+
+    shuffle : bool, optional, default True
+        Whether to shuffle samples in each iteration. Only used when
+        algorithm='sgd' or 'adam'.
+
+    tol : float, optional, default 1e-4
+        Tolerance for the optimization. When the loss or score is not improving
+        by more than tol for two consecutive iterations, convergence is
+        considered to be reached and the algorithm exits.
+
+    learning_rate_init : double, optional, default 0.001
+        The initial learning rate used. It controls the step-size
+        in updating the weights. Only used when algorithm='sgd' or 'adam'.
+
+    power_t : double, optional, default 0.5
+        The exponent for inverse scaling learning rate.
+        It is used in updating learning_rate_init when the learning_rate
+        is set to 'invscaling'. Only used when algorithm='sgd'.
+
+    verbose : bool, optional, default False
+        Whether to print progress messages to stdout.
+
+    warm_start : bool, optional, default False
+        When set to True, reuse the solution of the previous
+        call to fit as initialization, otherwise, just erase the
+        previous solution.
+
+    momentum : float, default 0.9
+        Momentum for gradient descent update.  Should be between 0 and 1. Only
+        used when algorithm='sgd'.
+
+    nesterovs_momentum : boolean, default True
+        Whether to use Nesterov's momentum. Only used when algorithm='sgd' and
+        momentum > 0.
+
+    early_stopping : bool, default False
+        Whether to use early stopping to terminate when loss is not improving.
+        If set to true, it will automatically set aside 10% of training data
+        as validation and termninate training when validation score is not
+        improving by more than tol for two consecutive epochs.
+        Only effective when algorithm='sgd' or 'adam'
+
+    validation_fraction : float, optional, default 0.1
```

Comment:
Should we also support sending in a slice object (or a KFold type iterator) so people can choose?